File: extract_info.js

const fs = require('fs');

function extractPois(filePath) {
    const content = fs.readFileSync(filePath, 'utf8');
    // Simple regex to find POI objects. This is not perfect but should work for this structure.
    const pois = [];
    const poiMatches = content.match(/\{[\s\S]*?id:[\s\S]*?\}\s*\},/g) || content.match(/\{[\s\S]*?"id":[\s\S]*?\}\s*\},/g);
    
    // Actually, let's use a more robust way by split or similar if possible.
    // Or just parse it if it was valid JSON, but it's a TS file with an array.
    
    // Let's try to find each POI block.
    const entries = content.split(/  \{/);
    entries.shift(); // remove imports part
    
    for (const entry of entries) {
        const idMatch = entry.match(/"id":\s*"([^"]+)"/) || entry.match(/id:\s*"([^"]+)"/);
        const nameHuMatch = entry.match(/"hu":\s*"([^"]+)"/) || entry.match(/hu:\s*"([^"]+)"/);
        const nameDeMatch = entry.match(/"de":\s*"([^"]+)"/) || entry.match(/de:\s*"([^"]+)"/);
        const typeMatch = entry.match(/"type":\s*"([^"]+)"/) || entry.match(/type:\s*"([^"]+)"/);
        
        if (idMatch) {
            pois.push({
                id: idMatch[1],
                nameHu: nameHuMatch ? nameHuMatch[1] : '',
                nameDe: nameDeMatch ? nameDeMatch[1] : '',
                type: typeMatch ? typeMatch[1] : ''
            });
        }
    }
    return pois;
}

const ro1 = extractPois('/mnt/c/Users/User/plizio-repo/lib/visualLab/data/poiExtraRo1.ts');
const ro2 = extractPois('/mnt/c/Users/User/plizio-repo/lib/visualLab/data/poiExtraRo2.ts');

console.log(JSON.stringify({ro1, ro2}, null, 2));

File: extract_info.py
import re
import json

def extract_pois(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    pois = content.split('id: "')
    result = []
    for poi in pois[1:]:
        poi_id = poi.split('"')[0]
        
        name_match = re.search(r'name:\s*\{\s*[^}]*en:\s*"([^"]+)"', poi)
        name = name_match.group(1) if name_match else ""
        
        en_desc_match = re.search(r'descriptionAdvanced:\s*\{[^}]*en:\s*"([^"]+)"', poi)
        en_desc = en_desc_match.group(1) if en_desc_match else ""
        
        if 'descriptionAdvanced: {\n      de: ""' in poi or 'descriptionAdvanced: {\n        de: ""' in poi or 'descriptionAdvanced: {\n      de: "",' in poi:
            result.append({"id": poi_id, "name": name, "en_desc": en_desc})
            
    return result

r1 = extract_pois('lib/visualLab/data/icelandPoi.ts')
r2 = extract_pois('lib/visualLab/data/poiExtraIcelandCities.ts')

with open('missing_pois.json', 'w', encoding='utf-8') as f:
    json.dump(r1 + r2, f, indent=2, ensure_ascii=False)
